testsuite/tests/translprim/ref_spec.ml

```diff
@@ -40,3 +40,13 @@ gen_rec.y <- D "foo";;
 gen_rec.y <- C;;
 flt_rec.y <- 1.;;
 flt_rec'.z <- 1.;;
+
+(* must use a write barrier, type is open *)
+let set_open_poly (r:[>`Foo] ref) y = r := y ;;
+let set_open_poly (r:[<`Foo] ref) y = r := y ;;
```

Comment:
sorry for the nitpicking, but could you use `set_open_poly_imm` or `set_open_poly_addr` in the name of the function, depending on whether or not you expect it to be optimized? This would make the lambda output much easier to read.